packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -110,7 +114,10 @@ export default function convertFunctionParams(path, loose) {
       });
       body.push(defNode);
     } else if (firstOptionalIndex !== null) {
-      const defNode = buildArgumentsAccess([param.node, t.numericLiteral(i)]);
+      const defNode =
+        param.isObjectPattern() || param.isArrayPattern()
```

Comment:
I'm planning to remove this check based on @bmeurer's feedback so please do not merge yet.